catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java

```diff
@@ -479,6 +479,41 @@ public SolrQuery propertyIsGreaterThanOrEqualTo(String propertyName, Date startD
         SOLR_INCLUSIVE_END);
   }
 
+  @Override
+  public SolrQuery propertyIsBetween(
+      String propertyName, short lowerBoundary, short upperBoundary) {
+    return getBetweenQuery(propertyName, AttributeFormat.SHORT, lowerBoundary, upperBoundary);
+  }
+
+  @Override
+  public SolrQuery propertyIsBetween(String propertyName, int lowerBoundary, int upperBoundary) {
+    return getBetweenQuery(propertyName, AttributeFormat.INTEGER, lowerBoundary, upperBoundary);
+  }
+
+  @Override
+  public SolrQuery propertyIsBetween(String propertyName, long lowerBoundary, long upperBoundary) {
+    return getBetweenQuery(propertyName, AttributeFormat.LONG, lowerBoundary, upperBoundary);
+  }
+
+  @Override
+  public SolrQuery propertyIsBetween(
+      String propertyName, float lowerBoundary, float upperBoundary) {
+    return getBetweenQuery(propertyName, AttributeFormat.FLOAT, lowerBoundary, upperBoundary);
+  }
+
+  @Override
+  public SolrQuery propertyIsBetween(
+      String propertyName, double lowerBoundary, double upperBoundary) {
+    return getBetweenQuery(propertyName, AttributeFormat.DOUBLE, lowerBoundary, upperBoundary);
+  }
+
+  @Override
+  public SolrQuery propertyIsBetween(
+      String propertyName, Object lowerBoundary, Object upperBoundary) {
+    return getBetweenQuery(
+        propertyName, AttributeFormat.FLOAT, (Number) lowerBoundary, (Number) upperBoundary);
```

Comment:
MOB: Check if it's an `instanceof` `Number` first, then possibly throw an illegal argument exception.